wgpu-hal/src/dx12/command.rs

```diff
@@ -254,18 +269,25 @@ impl crate::CommandEncoder<super::Api> for super::CommandEncoder {
     }
     unsafe fn discard_encoding(&mut self) {
         if let Some(list) = self.list.take() {
-            list.close();
-            self.free_lists.push(list);
+            if list.close().into_result().is_ok() {
+                self.free_lists.push(list);
+            } else {
+                list.destroy();
+            }
         }
     }
     unsafe fn end_encoding(&mut self) -> Result<super::CommandBuffer, crate::DeviceError> {
         let raw = self.list.take().unwrap();
-        raw.close();
-        Ok(super::CommandBuffer { raw })
+        let closed = raw.close().into_result().is_ok();
+        Ok(super::CommandBuffer { raw, closed })
     }
     unsafe fn reset_all<I: Iterator<Item = super::CommandBuffer>>(&mut self, command_buffers: I) {
         for cmd_buf in command_buffers {
-            self.free_lists.push(cmd_buf.raw);
+            if cmd_buf.closed {
+                self.free_lists.push(cmd_buf.raw);
+            } else {
+                cmd_buf.raw.close();
```

Comment:
If closing failed the first time, why would it succeed this time?

Reply:
I can't know whether commandbuffer has tried to close before, I checked the code again and found that commandbuffer is only constructed in end_encoding, we can indeed drop it directly.

Reply:
I should destroy it here.